packages/core/onClickOutside/index.ts

```diff
@@ -15,26 +16,44 @@ export interface OnClickOutsideOptions<E extends keyof OnClickOutsideEvents> ext
  * @param handler
  * @param options
  */
-export function onClickOutside<E extends keyof OnClickOutsideEvents = 'pointerdown'>(
+export function onClickOutside<E extends keyof OnClickOutsideEvents = 'pointerup'>(
   target: MaybeElementRef,
   handler: (evt: OnClickOutsideEvents[E]) => void,
   options: OnClickOutsideOptions<E> = {},
 ) {
-  const { window = defaultWindow, event = 'pointerdown' } = options
+  const { window = defaultWindow, event = 'pointerup' } = options
 
   if (!window)
     return
 
+  const shouldListen = ref(true)
+  const isUpEvent = /(click|up|end)/.test(event)
+
   const listener = (event: OnClickOutsideEvents[E]) => {
-    const el = unrefElement(target)
-    if (!el)
+    if (isUpEvent && !shouldListen.value)
       return
 
-    if (el === event.target || event.composedPath().includes(el))
+    const el = unrefElement(target)
+
+    if (!el || el === event.target || event.composedPath().includes(el))
       return
 
     handler(event)
   }
 
-  return useEventListener(window, event, listener, { passive: true })
+  const cleanup = [
+    useEventListener(window, event, listener, { passive: true }),
+  ]
+
+  if (isUpEvent) {
+    cleanup.push(
+      useEventListener(window, 'pointerdown', () => (shouldListen.value = true), { passive: true }),
+      useEventListener(window, 'pointermove', () => (shouldListen.value = false), { passive: true }),
+      useEventListener(window, 'scroll', () => (shouldListen.value = false), { passive: true }),
```

Comment:
I don't really feel good about this solution. For solving the original case #641, could we just note the caveat and ask users to wrap a parent div and use scroll in the child if they absolutely need that? Giving it works pretty well in most of the cases, I don't think adding the three additional listener is worthwhile for this edge cases.

Reply:
Hi, let me tell you history of this function and what bring us to here.

At the beginning of this function we were firing handler for `mousedown` and `touchstart` events. Because I wanted to call handler asap to make clicking outside feels optimistic. Then we added `pointerdown` for https://github.com/vueuse/vueuse/issues/323 . Now we are listening 3 events and calling handler multiple times for same event. This issue mentioned in https://github.com/vueuse/vueuse/issues/451 . Then I suggested setting default event to `pointerdown` on [my comment](https://github.com/vueuse/vueuse/issues/451#issuecomment-821197541). Because I thought the issue on https://github.com/vueuse/vueuse/issues/323 was that `mousedown` and `touchstart` events not firing with canvas. I was wrong. The issue was comes from line 40 of reproduction where creates `new OrbitControls` which is prevents firing of our event listeners. I totally missed that line and suggested defaulting to `pointerdown`.

This PR intended to fix #641 by listening `pointerup` event default, but then I realized few other issues like `pointerup` fires when selection event ends outside target, and provide workaround for it.

The biggest problem of this functions was not defaulting to `click` event at the beginning. This raised some issues like https://github.com/vueuse/vueuse/issues/681 and click event have not issue with selecting inline text, selecting input text or scrollbar clicks.

Defaulting `click` event is breaking chance at the moment and needs some refactors. When clicking **`Show Modal`** button also our event handler runs and immediately close modal. There is two way to fix that:

- Adding `.stop` modifier to click event on user land `@click.stop="modal = true"`. This is not preferred way to do for me since we are planning to defaulting `click` event.
- Register `click` listener when `target.value` becomes truthy. This is preferred way.

After defaulting `click` event I don't think anyone who needs other event listeners (like https://github.com/vueuse/vueuse/issues/323) will look for scrollbar click or better selection support. If they need that stuff, they can call another `onClickOutside`.

I'm totally fine with skipping the changes in this PR for v6. All of these workaround can be achieved in current version by setting `{ event: 'click' }` on `onClickOutside` and adding `.stop` modifier to template. I will make another PR for next major version.

Reply:
Thanks for the update. Looks like `click` is the way to go. I don't mind having a breaking change as I am also revising functions and working on the new docs recently (await for Vue's new docs).

If we go with `click`, I think we should probably not make it configurable in order to keep the function simple.

As for the clicking conflict, I guess we could solve that by adding `nextTick` somewhere, let's see the implementation first and I might be able to do some help.